src/main/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/service/BlobVerifierImpl.java

```diff
@@ -0,0 +1,133 @@
+package it.gov.pagopa.rtd.ms.rtdmsdecrypter.service;
+
+import static it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.BlobApplicationAware.Status.VERIFIED;
+
+import com.opencsv.bean.BeanVerifier;
+import com.opencsv.bean.CsvToBean;
+import com.opencsv.bean.CsvToBeanBuilder;
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.AdeTransactionsAggregate;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.BlobApplicationAware;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.BlobApplicationAware.Application;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.DecryptedRecord;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.RtdTransaction;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Implementation of the {@link BlobVerifier} interface, used to verify the validity of the
+ * {@link BlobApplicationAware} records extracted from the input file.
+ */
+@Service
+@Setter
+@Slf4j
+public class BlobVerifierImpl implements BlobVerifier {
+
+  @Value("${decrypt.skipChecksum}")
+  private boolean skipChecksum;
+
+  /**
+   * Verify method, used to verify the validity of the {@link BlobApplicationAware} records
+   * decrypted.
+   */
+  public BlobApplicationAware verify(BlobApplicationAware blob) {
+    log.info("START VERIFYING {}", blob.getBlob());
+
+    FileReader fileReader;
+
+    boolean isValid = true;
+
+    try {
+      fileReader = new FileReader(Path.of(blob.getTargetDir(), blob.getBlob()).toFile());
+    } catch (FileNotFoundException e) {
+      log.error("Error reading file {}", blob.getBlob());
+
+      return blob;
+    }
+
+    BeanVerifier<? extends DecryptedRecord> verifier = new RtdTransactionsVerifier();
```

Comment:
May be possible to change the object creation strategy? Don't you think it may be better to instantiate the Verifier classes as Bean? I imagine the scenario where for each file processed it will be created a new instance of the verifier (two if it's a ade file), rather than one instance for every file processed.

Reply:
I've seen the changes. Nice improve, good job! Thank you.

Reply:
Thank you for the observation!